### PR TITLE
Added render utilities

### PR DESCRIPTION
**Pull request by jojolepro**

### CODE REVIEW

File: amethyst_utils/src/render.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+//! Module containing utility methods to easily create `Material` and `Texture` handles using a minimal amount of code.
```
Comment (torkleyy) [inline amethyst_utils/src/render.rs]: The whole module seems to be about `albedo` only materials, this should be indicated somehow. Possibly in

* module path or
* documentation (or both)

Comment (jojolepro) [reply amethyst_utils/src/render.rs]: Currently its like that, but it will have more eventually. I just ported code that I already had on hand.

**Comment by Rhuagh** [comment]
Oh, and please add Progress parameters on the functions that create textures.

File: amethyst_utils/src/render.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#[derive(SystemData)]
+pub struct MaterialCreator<'a> {
+    loader: ReadExpect<'a, Loader>,
+    storage: Read<'a, AssetStorage<Texture>>,
+    defaults: ReadExpect<'a, MaterialDefaults>,
+}
+
+impl<'a> MaterialCreator<'a> {
+    /// Generate a `Material` from a color.
+    pub fn material_from_color<T: Progress>(
```
Comment (torkleyy) [inline amethyst_utils/src/render.rs]: Let's call this `material_from_albedo`

File: amethyst_utils/src/render.rs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        progress_counter: T,
+    ) -> Material {
+        let albedo = self
+            .loader
+            .load_from_data(color.into(), progress_counter, &self.storage);
+        self.material_from_texture(albedo)
+    }
+
+    /// Generate a `Material` from a path pointing to a png image.
+    pub fn material_from_png<T: Progress>(&self, path: String, progress_counter: T) -> Material {
```
Comment (torkleyy) [inline amethyst_utils/src/render.rs]: Could use `S: Into<String>`

File: amethyst_utils/src/render.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        color: [f32; 4],
+        progress_counter: T,
+    ) -> Material {
+        let albedo = self
+            .loader
+            .load_from_data(color.into(), progress_counter, &self.storage);
+        self.material_from_texture(albedo)
+    }
+
+    /// Generate a `Material` from a path pointing to a png image.
```
Comment (torkleyy) [inline amethyst_utils/src/render.rs]: Should specify the source used.

Comment (jojolepro) [reply amethyst_utils/src/render.rs]: file system path?

File: amethyst_utils/src/render.rs (hunk 2 above)
Comment (jojolepro) [reply amethyst_utils/src/render.rs]: Technically its material_from_albedo_from_color. Are you sure you want it called albedo?

**Comment by Rhuagh** [comment]
Looks good, just one nit: Some examples in the docs on usage would be stellar

File: amethyst_utils/src/render.rs (hunk 2 above)
Comment (torkleyy) [reply amethyst_utils/src/render.rs]: Dismissing...

**Comment by jojolepro** [comment]
ready

**Comment by jojolepro** [comment]
zzz this is going to miss the release because of the missing review.

File: amethyst_utils/src/render.rs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            path,
+            PngFormat,
+            TextureMetadata::srgb(),
+            progress_counter,
+            &self.storage,
+        ))
+    }
+
+    /// Generate a `Material` from a texture handle.
+    pub fn material_from_texture(&self, texture: Handle<Texture>) -> Material {
```
Comment (termhn) [inline amethyst_utils/src/render.rs]: I'm not sure how useful this actually is at the moment, it seems like just doing the contained code whenever you want a texture with an albedo is pretty simple... perhaps we could make this https://github.com/amethyst/amethyst/blob/master/amethyst_renderer/src/pass/util.rs#L24 a public part of the `mtl` module and then make this method `material_from_textures` something like:

```rs
pub fn material_from_textures(&self, textures: &[(TextureType, Handle<Texture>)] -> Material {
    ...
}
```
Which inserts the given textures into the correct slots and then uses default for the rest?

**Comment by jojolepro** [comment]
Waiting for rhuagh's review to squash.

**Comment by Rhuagh** [comment]
@jojolepro sorry for the holdup, lgtm

**Comment by jojolepro** [comment]
We'll let it build then we can merge :3

**Comment by jojolepro** [comment]
bors r+

you can do it!

**Comment by jojolepro** [comment]
yaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa yeet

**Pull request merged**